watchface/src/main/java/com/android/developers/androidify/watchface/creator/KeyGenerator.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.developers.androidify.watchface.creator
+
+import android.util.Base64
+import org.bouncycastle.asn1.x500.X500Name
+import org.bouncycastle.asn1.x500.style.BCStyle
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder
+import java.io.StringWriter
+import java.math.BigInteger
+import java.security.KeyPairGenerator
+import java.security.SecureRandom
+import java.util.Calendar
+import java.util.Date
+
+/**
+ * Generates a self-signed X.509 certificate and its private key. This is used to sign the APK.
+ *
+ * This is NOT the approach that should be taken in a production app.
```

Comment:
We should do whats recommended for production here :)

Reply:
Updated to use Android KeyStore. Requires some changes to Pack, which I've requested here: https://github.com/google/pack/pull/75

This ensures that the private key is never in the app's memory or at rest in app storage.